node/forest_libp2p/bitswap/tests/request_manager.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2019-2023 ChainSafe Systems
```

Comment:
That's a perfect test setup. It would be fantastic to have something like this on Forest-level.

Reply:
Obviously not in this PR, just a though I had when reviewing it and had to share. :)

Reply:
Yeah that will be great!